src/tests/intg/test_files_provider.py

```diff
@@ -0,0 +1,689 @@
+               gid=30012,
+               mem=['user1', 'user2'])
+
+group_nomem = dict(name='group_nomem',
+                   gid=40000,
+                   mem=[])
+
+
+def stop_sssd():
+    pid_file = open(config.PIDFILE_PATH, "r")
+    pid = int(pid_file.read())
+    os.kill(pid, signal.SIGTERM)
+    while True:
+        try:
+            os.kill(pid, signal.SIGCONT)
+        except:
+            break
+        time.sleep(1)
+
+
+def create_conf_fixture(request, contents):
+    """Generate sssd.conf and add teardown for removing it"""
+    conf = open(config.CONF_PATH, "w")
+    conf.write(contents)
+    conf.close()
+    os.chmod(config.CONF_PATH, stat.S_IRUSR | stat.S_IWUSR)
+    request.addfinalizer(lambda: os.unlink(config.CONF_PATH))
+
+
+def create_sssd_fixture(request):
+    """Start sssd and add teardown for stopping it and removing state"""
+    os.environ["SSS_FILES_PASSWD"] = os.environ["NSS_WRAPPER_PASSWD"]
+    os.environ["SSS_FILES_GROUP"] = os.environ["NSS_WRAPPER_GROUP"]
+    if subprocess.call(["sssd", "-D", "-f"]) != 0:
+        raise Exception("sssd start failed")
+
+    def teardown():
+        try:
+            stop_sssd()
+        except:
+            pass
+        for path in os.listdir(config.DB_PATH):
+            os.unlink(config.DB_PATH + "/" + path)
+        for path in os.listdir(config.MCACHE_PATH):
+            os.unlink(config.MCACHE_PATH + "/" + path)
+    request.addfinalizer(teardown)
+
+
+# Fixtures
+@pytest.fixture
+def files_domain_only(request):
+    conf = unindent("""\
+        [sssd]
+        domains             = files
+        services            = nss
+
+        [domain/files]
+        id_provider = files
+    """).format(**locals())
+    create_conf_fixture(request, conf)
+    create_sssd_fixture(request)
+    return None
+
+
+def setup_pw_with_list(request, user_list):
+    pwd_ops = passwd_ops_setup(request)
+    for user in user_list:
+        pwd_ops.useradd(**user)
+    ent.assert_passwd_by_name(canary['name'], canary)
+    return pwd_ops
+
+
+@pytest.fixture
+def add_user_with_canary(request):
+    return setup_pw_with_list(request, [canary, user1])
+
+
+@pytest.fixture
+def setup_pw_with_canary(request):
+    return setup_pw_with_list(request, [canary])
+
+
+def setup_gr_with_list(request, group_list):
+    grp_ops = group_ops_setup(request)
+    for group in group_list:
+        grp_ops.groupadd(**group)
+    ent.assert_group_by_name(canary_gr['name'], canary_gr)
+    return grp_ops
+
+
+@pytest.fixture
+def add_group_with_canary(request):
+    return setup_gr_with_list(request, [group1, canary_gr])
+
+
+@pytest.fixture
+def setup_gr_with_canary(request):
+    return setup_gr_with_list(request, [canary_gr])
+
+
+def poll_canary(fn, name, threshold=20):
+    for i in range(0, threshold):
+        res, _ = fn(name)
+        if res == NssReturnCode.SUCCESS:
+            return True
+        elif res == NssReturnCode.NOTFOUND:
+            time.sleep(0.1)
+            continue
```

Comment:
it might be good to add doc string here with explanation. Or to functions which uses this function.
IMHO, it would be easier to understand rather then in commit message